build.proj

```diff
@@ -190,214 +123,123 @@
+    <Exec Command="dotnet $(RepoArtifacts)StaticAnalysis/StaticAnalysis.Netcore.dll -p $(RepoArtifacts)$(Configuration) -r $(StaticAnalysisOutputDirectory) -h -u -m @(ModulesChanged)" Condition="'$(RunStaticAnalysis)' == 'true'" />
   </Target>
 
-  <!-- Run Validation -->
-  <Target Name="DependencyAnalysisNetcore"
-          Condition="'$(SkipDependencyAnalysis)' == 'false'">
-
-    <MSBuild  Projects="@(StaticAnalysisNetcore)"
-              Targets="Build"
-              Properties="Configuration=Debug;Platform=Any CPU"
-              ContinueOnError="false" />
+  <!-- Everything except Publish -->
+  <Target Name="Full" DependsOnTargets="Clean;Build;OnPremChecks;GenerateHelp;StaticAnalysis;Test" />
 
-    <Message Importance="high" Text="Running dependency analysis..." />
+  <Target Name="StaticAnalysis" Condition="'$(RunStaticAnalysis)' == 'true'">
+    <Message Importance="high" Text="Running static analysis..." />
+    <MakeDir Directories="$(StaticAnalysisOutputDirectory)" />
 
-    <PropertyGroup>
-        <SkipHelpValue Condition=" '$(SkipHelp)' == 'true' ">-s</SkipHelpValue>
-    </PropertyGroup>
-    <Exec Command="dotnet $(MSBuildProjectDirectory)\artifacts\StaticAnalysis.Netcore.dll -p $(MSBuildProjectDirectory)\artifacts\$(Configuration) -r $(MSBuildProjectDirectory)\artifacts $(SkipHelpValue) -u -m @(ModulesChanged)" />
-
-    <OnError ExecuteTargets="StaticAnalysisErrorMessage"/>
+    <Exec Command="dotnet $(RepoArtifacts)StaticAnalysis/StaticAnalysis.Netcore.dll -p $(RepoArtifacts)$(Configuration) -r $(StaticAnalysisOutputDirectory) -s -u -m @(ModulesChanged)" />
+    <Exec Command="$(PowerShellCoreCommandPrefix) &quot;. $(LibraryToolsFolder)/CheckAssemblies.ps1 -BuildConfig $(Configuration) &quot;" />
+    <OnError ExecuteTargets="StaticAnalysisErrorMessage" />
   </Target>
 
   <Target Name="StaticAnalysisErrorMessage">
     <Error Text="StaticAnalysis has failed. Please follow the instructions on this doc: https://github.com/Azure/azure-powershell/blob/master/documentation/Debugging-StaticAnalysis-Errors.md" />
   </Target>
 
   <Target Name="ChangeLogCheck">
-    <Exec Command="$(PowerShellCoreCommandPrefix) &quot;. $(LibraryToolsFolder)\CheckChangeLog.ps1 -FilesChanged '@(FilesChanged)'&quot;" ContinueOnError="false" Condition=" '$(Scope)' == 'All' or '$(Scope)' == 'Latest' " />
+    <Exec Command="$(PowerShellCoreCommandPrefix) &quot;. $(LibraryToolsFolder)/CheckChangeLog.ps1 -FilesChanged '@(FilesChanged)'&quot;" Condition=" '$(Scope)' == 'All' or '$(Scope)' == 'Latest' " />
     <OnError ExecuteTargets="ChangeLogErrorMessage" />
   </Target>
 
   <Target Name="ChangeLogErrorMessage">
     <Error Text="Modified files were found with no update to their change log. Please add a snippet to the affected modules' change log." />
   </Target>
 
-  <!-- Publish all packages -->
-  <Target Name="Publish">
+  <Target Name="Publish" Condition="'$(CodeSign)' == 'true'">
```

Comment:
I am not seeing where this target is called.  The only targets that mention this also appear not to be called from anywhere

Reply:
It is called as part of the sign build, since that is where we generate the nupkgs. Here's the sign build settings from my cloned Jenkins project:
![image](https://user-images.githubusercontent.com/17788297/51940250-5484d900-23c6-11e9-92cd-247f85fe5bec.png)